src/main/java/teammates/ui/controller/InstructorStudentListPageData.java

```diff
@@ -1,22 +1,63 @@
 package teammates.ui.controller;
 
-import java.util.HashMap;
+import java.util.ArrayList;
 import java.util.List;
+import java.util.Map;
 
 import teammates.common.datatransfer.AccountAttributes;
 import teammates.common.datatransfer.CourseAttributes;
-import teammates.common.datatransfer.InstructorAttributes;
+import teammates.ui.datatransfer.InstructorStudentListPageCourseData;
+import teammates.ui.template.InstructorStudentListFilterBox;
+import teammates.ui.template.InstructorStudentListFilterCourse;
+import teammates.ui.template.InstructorStudentListSearchBox;
+import teammates.ui.template.InstructorStudentListStudentsTableCourse;
 
 public class InstructorStudentListPageData extends PageData {
 
-    public InstructorStudentListPageData(AccountAttributes account) {
+    private InstructorStudentListSearchBox searchBox;
+    private InstructorStudentListFilterBox filterBox;
+    private List<InstructorStudentListStudentsTableCourse> studentsTable;
+    private int numOfCourses;
+    
+    public InstructorStudentListPageData(AccountAttributes account, String searchKey,
+                                         boolean displayArchive,
+                                         Map<String, String> numStudents,
+                                         List<InstructorStudentListPageCourseData> coursesToDisplay) {
         super(account);
+        this.searchBox = new InstructorStudentListSearchBox(getInstructorSearchLink(), searchKey, account.googleId);
+        List<InstructorStudentListFilterCourse> coursesForFilter =
+                                        new ArrayList<InstructorStudentListFilterCourse>();
+        List<InstructorStudentListStudentsTableCourse> coursesForStudentsTable =
+                                        new ArrayList<InstructorStudentListStudentsTableCourse>();
+        for (InstructorStudentListPageCourseData islpcData : coursesToDisplay) {
+            CourseAttributes course = islpcData.course;
+            coursesForFilter.add(new InstructorStudentListFilterCourse(course.id, course.name));
+            coursesForStudentsTable.add(
+                             new InstructorStudentListStudentsTableCourse(islpcData.isCourseArchived, course.id,
+                                                                          course.name, account.googleId,
+                                                                          numStudents.get(course.id),
+                                                                          getInstructorCourseEnrollLink(course.id),
+                                                                          islpcData.isInstructorAllowedToModify));
```

Comment:
Maybe just use two or three lines here instead?
Also, the line 36's indentation level is slightly off too.